### PR TITLE
Consider Gopkg.lock and glide.lock as generated

### PR DESCRIPTION
**Pull request by burke**
Recognize [`Gopkg.lock`](https://golang.github.io/dep/) and [`glide.lock`](https://glide.sh/) — two common Go package manager lockfiles — as generated.

## Checklist:
- [x] **I am adding new or changing current functionality**
  
  - [x] I have added or updated the tests for the new or changed functionality.

### CODE REVIEW

File: samples/Go/filenames/glide.lock [hunk 1]
```diff
@@ -0,0 +1,10 @@
+hash: e7e387e385240ea7d48e6de6a0c58fe523bfefa1d7895a2c25c100feb978774d
```
Comment (pchaigno) [inline samples/Go/filenames/glide.lock]: Is this format YAML?

File: samples/Go/filenames/Gopkg.lock [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# This file is autogenerated, do not edit; changes may be undone by the next 'dep ensure'.
```
Comment (pchaigno) [inline samples/Go/filenames/Gopkg.lock]: What format is this file?

**Comment by burke** [comment]
Ah, I completely misunderstood how the tests worked. Revised.

File: lib/linguist/languages.yml [hunk 3]
```diff
@@ -5273,6 +5274,7 @@ YAML:
   - ".clang-format"
   - ".clang-tidy"
   - ".gemrc"
+  - "glide.lock"
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: I'm clearly nitpicking here, but the double quotes are not required when the name doesn't start with a dot (we try to follow Ruby's YAML format).

**Pull request merged**